tests/serverpod_test_server/test_integration/method_streaming_cancel_test.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:test/test.dart';
+
+import 'test_tools/serverpod_test_tools.dart';
+
+void main() {
+  withServerpod(
+    'Given call to MethodStreamingEndpoint',
+    (endpoints, session) {
+      test(
+          'when calling an endpoint returning a non-broadcast stream and cancelling '
+          'then will cancel', () async {
+        var stream = endpoints.methodStreaming.intStreamFromValue(session, 1);
+        var subscription = stream.listen((event) {});
+
+        await expectLater(subscription.cancel(), completes);
+      });
+
+      test(
+        'when calling an endpoint returning a broadcast stream and cancelling '
+        'then will cancel',
+        () async {
+          var stream = endpoints.methodStreaming.getBroadcastStream(session);
+          var subscription = stream.listen((event) {});
+
+          await expectLater(subscription.cancel(), completes);
```

Comment:
@SandPod This test was throwing the reported error (see below) before the fix was applied. So the current assertion should be enough and I'm not convinced we need more. What we could do is as we discussed, namely add some variable to the endpoint that is set upon cancel to ensure it was executed as intended. However, this expands the scope of what we consider "correct" to be. By the same reasoning we should also then ensure the `session` was closed etc. WDYT?
```
type '() => void' is not a subtype of type '(() => Future<Null>?)?' of 'onTimeout'
```

Reply:
The test is probably enough!

Do we think it is enough that the test is applied through our test framework? Might be good enough?

Reply:
I think we could consider it enough, since we are (at least trying to) guaranteeing that the test tool should take the same code path; in this case the MethodStreamManager behind the scenes.

Reply:
Did you have any idea for alternative?

Reply:
Not really, we should guarantee this on the Server side so this should be a concern there.

Since the issue was specifially broadcast stream related I think it makes sense to just make sure that codepath is exercided on the Server somewhere. Let's go with this!

Reply:
Found a bug in the test stream manager that was fixed. I also extended the tests a bit to ensure everything is cleaned up properly: 
23e28bd